### PR TITLE
Feature/sign in user

### PR DESCRIPTION
**Pull request by agustinaGallo97**
## :link: Jira Tracker reference:
*As a user I want to sign in to can navigate the app  - https://androidtraining.atlassian.net/browse/AT-71

## Description:
Sign in created

## :pushpin: Notes:
The user can go to create an accout from this screen.

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -5,44 +5,45 @@
...
         android:label="@string/app_name"
         android:roundIcon="@mipmap/ic_launcher_round"
         android:supportsRtl="true"
         android:theme="@style/AppTheme"
         >
-
         <activity
-            android:name="com.xmartlabs.template.ui.login.LoginActivity"
-            android:label="@string/login"
+            android:name=".ui.onboarding.signIn.SignInActivity"
```
Comment (FerrariBruno) [inline app/src/main/AndroidManifest.xml]: package name should be all lowercase

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -5,44 +5,45 @@
...
+            android:screenOrientation="portrait"
+            android:theme="@style/AppTheme.TranslucentStatusBar"
+            ></activity>
```
Comment (FerrariBruno) [inline app/src/main/AndroidManifest.xml]:
```suggestion
            >
            
            </activity>
```

File: app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -5,44 +5,45 @@
...
+        <activity
+            android:name=".ui.onboarding.signUp.SignUpActivity"
```
Comment (FerrariBruno) [inline app/src/main/AndroidManifest.xml]: same

File: app/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -5,44 +5,45 @@
...
+            android:label="Signup"
             android:screenOrientation="portrait"
             android:theme="@style/AppTheme.TranslucentStatusBar"
             />
-
         <activity
-            android:name="com.xmartlabs.template.ui.onboarding.OnboardingActivity"
-            android:label="@string/onboarding"
+            android:name=".ui.login.LoginActivity"
```
Comment (FerrariBruno) [inline app/src/main/AndroidManifest.xml]: what's the difference between this activity and `".ui.onboarding.signIn.SignInActivity"` declared above?

File: app/src/main/AndroidManifest.xml [hunk 5]
```diff
@@ -5,44 +5,45 @@
...
 
                 <category android:name="android.intent.category.LAUNCHER" />
             </intent-filter>
-
         </activity>
-
     </application>
 
-</manifest>
+</manifest>
```
Comment (FerrariBruno) [inline app/src/main/AndroidManifest.xml]:
```suggestion
</manifest>

```

File: app/src/main/java/com/xmartlabs/template/di/FragmentBuildersModule.kt [hunk 6]
```diff
@@ -14,11 +14,11 @@ abstract class FragmentBuildersModule {
   abstract fun contributeLoginFragment(): LoginFragment
 
   @ContributesAndroidInjector
-  abstract fun contributeOnboardingFragment(): OnboardingFragment
+  abstract fun contributeRecyclerExampleFragment(): RecyclerExampleFragment
```
Comment (FerrariBruno) [inline app/src/main/java/com/xmartlabs/template/di/FragmentBuildersModule.kt]: why?

File: app/src/main/AndroidManifest.xml (hunk 4 above)
Comment (FerrariBruno) [reply app/src/main/AndroidManifest.xml]: also, why remove the onboarding activity?

File: app/src/main/java/com/xmartlabs/template/helper/OnBoardingValidator.kt [hunk 7]
```diff
@@ -0,0 +1,16 @@
+package com.xmartlabs.template.helper
+
+object OnBoardingValidator {
```
Comment (FerrariBruno) [inline app/src/main/java/com/xmartlabs/template/helper/OnBoardingValidator.kt]: should be `OnboardingValidator` as `Onboarding` is only one word

File: app/src/main/java/com/xmartlabs/template/ui/StartActivity.kt [hunk 8]
```diff
@@ -15,16 +15,17 @@ class StartActivity : BaseAppCompatActivity() {
   override fun onCreate(savedInstanceState: Bundle?) {
     super.onCreate(savedInstanceState)
 
-    val session = sessionRepository.session
-    if (session == null) {
-      // TODO: handle on start activity no session
+//    val session = sessionRepository.session
```
Comment (FerrariBruno) [inline app/src/main/java/com/xmartlabs/template/ui/StartActivity.kt]: why? I think you'll need this. I agree that maybe not now, but you should mark this with a TODO or something as we usually don't leave any commented code

File: app/src/main/java/com/xmartlabs/template/ui/onboarding/signUp/SignUpFragment.kt [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+              !OnBoardingValidator.isCorrectPassword(password.text.toString()) -> R.string.error_invalid_password
+              else -> R.string.error_check_password
+            }
+        ), Toast.LENGTH_SHORT).show()
+      }
+    }
+  }
+
+  private fun isCorrectRequest(): Boolean =
+      (OnBoardingValidator.isCorrectName(user.text.toString()) &&
```
Comment (FerrariBruno) [inline app/src/main/java/com/xmartlabs/template/ui/onboarding/signUp/SignUpFragment.kt]: parentheses not needed

File: app/src/main/java/com/xmartlabs/template/usecase/BaseSingleUseCase.kt [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+import androidx.lifecycle.MutableLiveData
+import com.xmartlabs.template.usecase.internal.DefaultScheduler
+import com.xmartlabs.template.usecase.internal.Scheduler
+import io.reactivex.Single
+import timber.log.Timber
+
+/*
+ This class is created as a template of each necessary useCase.
+*/
+
```
Comment (FerrariBruno) [inline app/src/main/java/com/xmartlabs/template/usecase/BaseSingleUseCase.kt]: maybe include the source of this class

File: app/src/main/java/com/xmartlabs/template/usecase/internal/TaskScheduler.kt [hunk 11]
```diff
@@ -0,0 +1,109 @@
+package com.xmartlabs.template.usecase.internal
+
+/*
+ This class is created to control the handle of operations.
+*/
+
```
Comment (FerrariBruno) [inline app/src/main/java/com/xmartlabs/template/usecase/internal/TaskScheduler.kt]: same

File: app/src/main/res/drawable/gradient.xml [hunk 12]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="utf-8"?>
+<selector xmlns:android="http://schemas.android.com/apk/res/android">
+
+    <item>
```
Comment (FerrariBruno) [inline app/src/main/res/drawable/gradient.xml:4]:
```suggestion
    <item>
    
```

File: app/src/main/res/drawable/gradient.xml [hunk 13]
```diff
@@ -0,0 +1,16 @@
...
+        <shape>
```
Comment (FerrariBruno) [inline app/src/main/res/drawable/gradient.xml:6]:
```suggestion
        <shape>
        
```

File: app/src/main/res/drawable/gradient.xml [hunk 14]
```diff
@@ -0,0 +1,16 @@
...
+            <gradient
+                android:angle="90"
+                android:endColor="#ff8256"
+                android:shape="oval"
+                android:startColor="#9d0066"
+                android:type="linear"
+                />
```
Comment (FerrariBruno) [inline app/src/main/res/drawable/gradient.xml:14]:
```suggestion
                />
                
```

File: app/src/main/res/drawable/gradient.xml [hunk 15]
```diff
@@ -0,0 +1,16 @@
...
+        </shape>
```
Comment (FerrariBruno) [inline app/src/main/res/drawable/gradient.xml:16]:
```suggestion
        </shape>
        
```

File: app/src/main/res/drawable/on_boarding_shape.xml [hunk 16]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="utf-8"?>
+<layer-list
```
Comment (FerrariBruno) [inline app/src/main/res/drawable/on_boarding_shape.xml]: add spaces between all elements

File: app/src/main/res/drawable/on_boarding_shape.xml [hunk 17]
```diff
@@ -0,0 +1,25 @@
...
+        <shape android:shape="rectangle">
+            <size android:height="45dp" />
+            <solid android:color="#ffffff" />
+            <corners
+                android:bottomRightRadius="10dp"
+                android:topRightRadius="10dp"
+                />
+        </shape>
+    </item>
+</layer-list>
```
Comment (FerrariBruno) [inline app/src/main/res/drawable/on_boarding_shape.xml]:
```suggestion
</layer-list>

```

File: app/src/main/res/layout/fragment_sign_in.xml [hunk 18]
```diff
@@ -0,0 +1,121 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:id="@+id/linearLayout"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:background="@drawable/gradient"
```
Comment (FerrariBruno) [inline app/src/main/res/layout/fragment_sign_in.xml]: sort

File: app/src/main/res/layout/fragment_signup.xml [hunk 19]
```diff
@@ -0,0 +1,151 @@
...
+        android:layout_marginStart="118dp"
+        android:layout_marginTop="29dp"
+        android:text="@string/create_account"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintLeft_toLeftOf="parent"
+        app:layout_constraintRight_toRightOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/checkPasswordTextField"
+        style="@style/ButtonStyle"
+        />
+    />
```
Comment (FerrariBruno) [inline app/src/main/res/layout/fragment_signup.xml]: is this compiling?

File: app/src/main/res/values/strings.xml [hunk 20]
```diff
@@ -22,4 +22,20 @@
...
     <string name="username">Username</string>
     <string name="password">Password</string>
+    <string name="user">User</string>
+    <string name="mail">Mail</string>
+    <string name="password_check">Check password</string>
+    <string name="create_account">Create account</string>
+    <string name="title">Taskloans</string>
+    <string name="sign_up_correctly">Signed up correctly</string>
+    <string name="sign_in_correctly">Correct login</string>
+    <string name="sign_in">Entrar</string>
```
Comment (FerrariBruno) [inline app/src/main/res/values/strings.xml]: shouldn't it be in English?

File: app/src/main/AndroidManifest.xml (hunk 4 above)
Comment (agustinaGallo97) [reply app/src/main/AndroidManifest.xml]: It is LoginActivity belongs to base project

File: app/src/main/java/com/xmartlabs/template/di/FragmentBuildersModule.kt (hunk 6 above)
Comment (agustinaGallo97) [reply app/src/main/java/com/xmartlabs/template/di/FragmentBuildersModule.kt]: Like above I deleted some classes that belong to the base project

File: app/src/main/java/com/xmartlabs/template/helper/OnBoardingValidator.kt (hunk 7 above)
Comment (f7deleon) [reply app/src/main/java/com/xmartlabs/template/helper/OnBoardingValidator.kt]: i don't think an object is needed here maybe just a file?

File: app/build.gradle [hunk 21]
```diff
@@ -179,6 +179,8 @@ final TIMBER_VERSION = '4.7.1'
 final TRACEUR_VERSION = '273cc7d582'
 
 dependencies {
+    implementation 'androidx.appcompat:appcompat:1.1.0'
```
Comment (FerrariBruno) [inline app/build.gradle]: please add versions above like `final TRACEUR_VERSION = '273cc7d582'`

File: app/src/main/java/com/xmartlabs/taskloans/ui/onboarding/signin/SignInFragment.kt [hunk 22]
```diff
@@ -0,0 +1,85 @@
...
+  private fun signInViewModelObserver() {
+    signInViewModel.signIn.observe(this, Observer { result ->
+      when {
+        result.isSuccess -> onSuccessResult()
+        result.isFailure -> onFailureResult()
+      }
+    })
+  }
+
+  private fun onFailureResult() {
```
Comment (FerrariBruno) [inline app/src/main/java/com/xmartlabs/taskloans/ui/onboarding/signin/SignInFragment.kt]: I think you can use `=`

File: app/src/main/java/com/xmartlabs/taskloans/ui/onboarding/signin/SignInFragment.kt [hunk 23]
```diff
@@ -0,0 +1,85 @@
...
+    Toast.makeText(context, App.context.resources.getString(R.string.error_incorrect_login), Toast.LENGTH_SHORT)
+        .show()
+  }
+
+  private fun onSuccessResult() {
```
Comment (FerrariBruno) [inline app/src/main/java/com/xmartlabs/taskloans/ui/onboarding/signin/SignInFragment.kt]: same

File: app/src/main/java/com/xmartlabs/taskloans/ui/onboarding/signin/SignInFragment.kt [hunk 24]
```diff
@@ -0,0 +1,85 @@
...
+    // TODO fix it in the future
+    Toast.makeText(context, App.context.resources.getString(R.string.sign_in_correctly), Toast.LENGTH_SHORT)
+        .show()
+  }
+
+  private fun buttonsListener() {
```
Comment (FerrariBruno) [inline app/src/main/java/com/xmartlabs/taskloans/ui/onboarding/signin/SignInFragment.kt]: maybe something like `setupListeners`?

File: app/src/main/java/com/xmartlabs/taskloans/ui/onboarding/signin/SignInFragment.kt [hunk 25]
```diff
@@ -0,0 +1,85 @@
...
+    loginButtonListener()
+    createAccountTextViewListener()
+  }
+
+  private fun loginButtonListener() {
```
Comment (FerrariBruno) [inline app/src/main/java/com/xmartlabs/taskloans/ui/onboarding/signin/SignInFragment.kt]: same comment as above

File: app/src/main/java/com/xmartlabs/taskloans/ui/onboarding/signin/SignInFragment.kt [hunk 26]
```diff
@@ -0,0 +1,85 @@
...
+            when {
+              !isCorrectMail(mailTextInputEditText.text.toString()) -> R.string.error_invalid_mail
+              else -> R.string.error_invalid_password
+            }
+        ), Toast.LENGTH_SHORT).show()
+      }
+    }
+  }
+
+  private fun createAccountTextViewListener() {
```
Comment (FerrariBruno) [inline app/src/main/java/com/xmartlabs/taskloans/ui/onboarding/signin/SignInFragment.kt]: you can use `=`

File: app/src/main/java/com/xmartlabs/taskloans/ui/onboarding/signin/SignInFragment.kt (hunk 25 above)
Comment (FerrariBruno) [reply app/src/main/java/com/xmartlabs/taskloans/ui/onboarding/signin/SignInFragment.kt]: also you can use `=`

File: app/src/main/java/com/xmartlabs/taskloans/ui/onboarding/signin/SignInFragment.kt [hunk 27]
```diff
@@ -0,0 +1,85 @@
...
+    createAccountTextView.setOnClickListener {
+      val intent = Henson.with(context)
+          .gotoSignUpActivity()
+          .build()
+          .setFlags(Intent.FLAG_ACTIVITY_NO_ANIMATION or Intent.FLAG_ACTIVITY_CLEAR_TASK)
+      startActivity(intent)
+    }
+  }
+
+  private fun isCorrectRequest(): Boolean =
```
Comment (FerrariBruno) [inline app/src/main/java/com/xmartlabs/taskloans/ui/onboarding/signin/SignInFragment.kt]: type declaration shouldn't be necessary here

**Review by FerrariBruno — APPROVED** [review]

**Pull request merged**